lib/smart_name.rb

```diff
@@ -26,7 +33,12 @@ class SmartName < Object
   class << self
     def new obj
       return obj if obj.is_a? self.class
-      str = obj.is_a?(Array) ? obj * joint : obj.to_s
+      str =
+        if obj.is_a?(Array)
+          obj.map { |i| i.to_s } * joint
```

Comment:
Use map(&:to_s) syntax, it makes this way cleaner and still suitable for one line conditional if you wanted.